frontend/src/components/SearchBar.tsx

```diff
@@ -0,0 +1,260 @@
+import * as React from "react";
+import { useState, useEffect, useRef } from "react";
+import { useTranslation } from "react-i18next";
+import { useHistory } from "react-router-dom";
+
+//material ui
+import OutlinedInput from "@mui/material/OutlinedInput";
+import MenuItem from "@mui/material/MenuItem";
+import FormControl from "@mui/material/FormControl";
+import Select from "@mui/material/Select";
+import { makeStyles, Typography } from "@material-ui/core";
+import { Button } from "@material-ui/core";
+import Paper from "@material-ui/core/Paper";
+import ListItemText from "@mui/material/ListItemText";
+import Checkbox from "@mui/material/Checkbox";
+import TextField from "@mui/material/TextField";
+
+//project resources
+import theme from "../util/theme";
+import categories from "../util/categories";
+import { IChain } from "../types";
+
+interface IProps {
+  data: any;
+  setData: any;
+  setViewport: any;
```

Comment:
Might be nice to actually use types for this. Especially for parameters like this it matters.